src/AgentCommerce/HoneyWebhookService.php

```diff
@@ -0,0 +1,178 @@
+            $this->systemConfigService->set(Settings::AGENT_COMMERCE_ONBOARDED, false, $salesChannelId);
+        }
+
+        return $result;
+    }
+
+    private function createToken(string $salesChannelId, Context $context): ?string
+    {
+        $salesChannel = $this->loadSalesChannel($salesChannelId, $context);
+        if (!$salesChannel || !$salesChannel->getActive()) {
+            return null;
+        }
+
+        $productExport = $salesChannel->getProductExports()?->first();
+        $storefront = $productExport?->getStorefrontSalesChannel();
+        if (!$storefront) {
+            return null;
+        }
+
+        $url = $storefront->getHreflangDefaultDomain()?->getUrl() ?? $storefront->getDomains()?->first()?->getUrl();
+
+        $routes = $this->router->getRouteCollection()->get('store-api.product.export');
+        if (!$routes) {
+            return null;
+        }
+
+        $path = str_replace(['{accessKey}', '{fileName}'], [$productExport->getAccessKey(), $productExport->getFileName()], $routes->getPath());
+
+        $tokenBuilder = Builder::new(new JoseEncoder(), ChainedFormatter::default());
+
+        return $tokenBuilder
+            ->withClaim('storeName', $salesChannel->getName())
+            ->withClaim('storeUrl', $url)
+            ->withClaim('country', $salesChannel->getCountry()?->getIso())
+            ->withClaim('currency', $salesChannel->getCurrency()?->getIsoCode())
+            ->withClaim('favIcon', 'https://localhost/favicon.ico') // TODO: Need to be load
+            ->withClaim('shippingCountries', $storefront->getCountries()?->map(fn (CountryEntity $country) => $country->getIso()))
+            ->withClaim('paypalMerchantId', $this->credentialsUtil->getMerchantPayerId($storefront->getId()))
+            ->withClaim('shopwareMerchantId', $salesChannel->getId())
+            ->withClaim('catalogDownloadUrl', rtrim($url ?? '', '/') . $path)
+            ->getToken(new Sha256(), InMemory::plainText(random_bytes(32)))
+            ->toString();
+    }
+
+    private function loadSalesChannel(string $salesChannelId, Context $context): ?SalesChannelEntity
+    {
+        $criteria = new Criteria([$salesChannelId]);
+        $criteria->addFilter(new EqualsFilter('typeId', SwagPayPal::SALES_CHANNEL_TYPE_AGENT_COMMERCE));
+        $criteria->addAssociations([
+            'country',
+            'currency',
+            'domains',
+            'productExports.storefrontSalesChannel.domains',
+            'productExports.storefrontSalesChannel.hreflangDefaultDomain',
+            'productExports.storefrontSalesChannel.countries',
+        ]);
+
+        return $this->salesChannelRepository->search($criteria, $context)->first();
+    }
+
+    /**
+     * @return array{success: bool, message: string, error?: string}
+     */
+    private function webhookCall(string $salesChannelId, string $endpoint, Context $context): array
+    {
+        $token = $this->createToken($salesChannelId, $context);
```

Comment:
The token creation can fail for various reasons, but we only log "naa its missing". This can be confusing when debugging this :thinking: